crates/stc_ts_type_checker/tests/conformance/types/literal/stringMappingOverPatternLiterals.stats.rust-debug

```diff
@@ -1,6 +1,6 @@
 Stats {
-    required_error: 2,
-    matched_error: 27,
+    required_error: 6,
+    matched_error: 23,
```

Comment:
Regression here is because I haven't implemented a way yet to figure out whether we're in a variable assignment

```ts
let x: Uppercase<string>;
let y: Lowercase<string> = 'lower'
x = y // should be error
```
or a type alias assignment
```ts
type A = Uppercase<string> // should be no error
```
(and I think I might need to differentiate for type params, but not 100% sure yet)
```ts
type X<T extends string> = Uppercase<T>;
type S = X<Lowercase<string>> // no error
```

Reply:
Ended up doing https://github.com/dudykr/stc/blob/6fd7e1db7630dff38f0f8d4e5d5badee4ec15250/crates/stc_ts_file_analyzer/src/analyzer/convert/mod.rs#L284-L292

to differentiate. Not sure if `in_actual_type` is the correct choice but I want to avoid creating another property without being sure this is the correct approach.

Reply:
I think validating an alias only if it's a specific type is a bad idea.

Reply:
I think it should instead be 

```rust
self.normalize(span, alias, Default::default).context("tried to normalize type alias").report(&mut self.stroage);
```

Reply:
FWIW I tried this, but it doesn't report any errors for type aliases :cold_sweat:

Reply:
If so, I think you should add logic to https://github.com/dudykr/stc/blob/bfeb6fbb6211618e1165eca7710186d6427b2c8e/crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs#L1964-L2112

Reply:
I might be missing something, but I dont think this approach will work because I'm currently relying on `ctx` which doesn't seem to be passed on `self.normalize`

ex:
(doing this because I need to know if its a type alias assignment or not)
```
        let ctx = Ctx {
            in_actual_type: true,
            ..self.ctx
        };

        self.with_ctx(ctx)
            .normalize(Some(d.span), Cow::Borrowed(&alias), Default::default())?;
            
        // self.ctx.in_actual_type -> false inside the expand_intrinsic_types function
```

What I had thought of initially was a generic function to validate the rhs of the type alias assignment, because this problem doesn't seem unique to string intrinsic types. Im not sure what that would like exactly.

Reply:
Ended up with 

```rs
if let Err(e) = self.normalize(Some(d.span), Cow::Borrowed(&alias), Default::default()) {
            return Err(e);
        }

        let ctx = Ctx {
            in_actual_type: true,
            ..self.ctx
        };

        if let Type::StringMapping(ty) = alias.normalize() {
            if let Err(e) = self
                .with_ctx(ctx)
                .assign_to_intrinsic(&mut Default::default(), ty, &ty.type_args.params[0], Default::default())
            {
                return Err(e);
            }
        }
```

This is done because I cant pass ctx in `.normalize`. I considered creating `self.validate_rhs_of_alias` but I think at this point Idk if we need something like that.

Extra errors and panics due to calling `self.normalize`

Reply:
You can add a field to `NormalizeOpts`, or do `self.with_ctx(ctx).normalize(...)`

Reply:
I'm really not sure what to make of this PR, calling `assign_to_intrinsic` causes so many side effects I eventually gave up on making it work. I'm unsure whether that's caused by my code (partially it was I had to do some fixes), or whether having `assign_to_intrinsic` inside `normalize` is problematic.

Perhaps this PR should be closed/blocked until type-checking for type aliases is implemented in some way?

Reply:
Hmm.. I'll try implementing validation for type alias soon.

Reply:
I did it 😄

Reply:
Oh yeah thanks a lot for unblocking me :pray: , I've been working on top of your changes for last couple of hours. It seems likely that this will work out :crossed_fingers: